tests/ops/op_math/test_controlled.py

```diff
@@ -2183,8 +2184,8 @@ def ansatz(weights, wires):
 
         tape = QuantumScript.from_queue(q_tape)
         tape = tape.expand(depth=1, stop_at=lambda obj: not isinstance(obj, Controlled))
-        assert len(tape.operations) == 10
-        assert all(o.name in {"CNOT", "CRX", "Toffoli"} for o in tape.operations)
+        assert len(tape.operations) == 12
+        assert all(o.name in {"ControlledPhaseShift", "CRX", "Toffoli"} for o in tape.operations)
```

Comment:
Just to make sure I understand: The requirement for a decomposition is not to imply any ordering of "complex gates" to "simpler gates", but just should allow us to keep decomposing until external stopping criteria (like those of a device) trigger? :)

Reply:
Exactly!